### ISSUE TITLE
About lambda_st

### ISSUE THREAD

**Issue by wnma3mz** [AberHu/Knowledge-Distillation-Zoo#1]
Thank you very much for your project and it has helped me a lot. But I have some doubts about `train_st.py`.

https://github.com/AberHu/Knowledge-Distillation-Zoo/blob/b7bce273d9ee85331a75f5423d7b65c8834b04df/train_st.py#L184-L188

Here `lambda_st` is similar to a relaxation factor. But I see the following formula elsewhere. 

```python
 cls_loss = criterionCls(output_s, target) 
 st_loss  = criterionST(F.log_softmax(output_s/args.T, dim=1), 
 	                   F.softmax(output_t/args.T, dim=1)) * (args.T*args.T) / img.size(0) 

 loss = (1 - * args.lambda_st ) * cls_loss + st_loss * args.lambda_st 
```

Could you tell me the impact of the two?

https://github.com/PolarisShi/distillation/blob/8290b4b2138410cb4f1f39ecf2411f7dbb2a676a/student2.py#L218-L223

**Comment by Ga-Lee** [comment]
> Thank you very much for your project and it has helped me a lot. But I have some doubts about `train_st.py`.
> 
> https://github.com/AberHu/Knowledge-Distillation-Zoo/blob/b7bce273d9ee85331a75f5423d7b65c8834b04df/train_st.py#L184-L188
> 
> Here `lambda_st` is similar to a relaxation factor. But I see the following formula elsewhere.
> 
> ```python
>  cls_loss = criterionCls(output_s, target) 
>  st_loss  = criterionST(F.log_softmax(output_s/args.T, dim=1), 
>  	                   F.softmax(output_t/args.T, dim=1)) * (args.T*args.T) / img.size(0) 
> 
>  loss = (1 - * args.lambda_st ) * cls_loss + st_loss * args.lambda_st 
> ```
> 
> Could you tell me the impact of the two?
> 
> https://github.com/PolarisShi/distillation/blob/8290b4b2138410cb4f1f39ecf2411f7dbb2a676a/student2.py#L218-L223

hello, I feel a little confused about it ,too. May you explain it to me? Thanks a lot

**Comment by wnma3mz** [comment]
@Ga-Lee  My own guess is that it depends more on the loss of ground truth. `st_loss` exists as a constraint. Either should work, but it may depend on the specific downstream task.

**Comment by Ga-Lee** [comment]
> @Ga-Lee My own guess is that it depends more on the loss of ground truth. `st_loss` exists as a constraint. Either should work, but it may depend on the specific downstream task.

Got that, thank you~
